### PR TITLE
PLATUI-552: Updated frontend microservice creation to remove play-ui,…

### PR DESCRIPTION
**Pull request by JoPintoPaul**
… add play-frontend-hmrc and play-frontend-govuk

### CODE REVIEW

File: templates/service/test/controllers/LanguageSwitchControllerSpec.scala [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      cookies(result).get("PLAY_LANG").isDefined shouldBe false
+    }
+
+    "set the PLAY_LANG cookie correctly for English" in new {
+      val result = switchToEnglish(buildAppWithWelshLanguageSupport())
+      cookies(result).get("PLAY_LANG").isDefined shouldBe true
+      cookies(result).get("PLAY_LANG").get.value shouldBe "en"
+    }
+
+    "redirect to the cookie settings page by default" in new {
```
Comment (matthewmascord) [inline templates/service/test/controllers/LanguageSwitchControllerSpec.scala:61]: That will need changing.

Comment (JoPintoPaul) [reply templates/service/test/controllers/LanguageSwitchControllerSpec.scala]: Fixed.

File: templates/service/build.sbt [hunk 2]
```diff
@@ -12,12 +12,18 @@ lazy val microservice = Project(appName, file("."))
...
+      "uk.gov.hmrc.govukfrontend.views.html.helpers._",
+      "uk.gov.hmrc.hmrcfrontend.views.html.components._"
+    ),
     // ***************
     // Use the silencer plugin to suppress warnings
     <!--(if type=="FRONTEND")-->
     // You may turn it on for `views` too to suppress warnings from unused imports in compiled twirl templates, but this will hide other warnings.
     <!--(end)-->
-    scalacOptions += "-P:silencer:pathFilters=routes",
+    scalacOptions += "-P:silencer:pathFilters=routes;views",
```
[the remark this answers is not in the document]
Comment (JoPintoPaul) [reply templates/service/build.sbt]: Good point regarding the comment above. I've reverted this change.

File: templates/service/app/controllers/LanguageSwitchController.scala [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package uk.gov.hmrc.$!APP_PACKAGE_NAME!$.controllers
+
+import com.google.inject.Inject
+import javax.inject.Singleton
+import play.api.Configuration
+import play.api.i18n.Lang
+import play.api.mvc._
```
[the remark this answers is not in the document]
Comment (JoPintoPaul) [reply templates/service/app/controllers/LanguageSwitchController.scala]: Removed this import.

File: templates/service/app/controllers/LanguageSwitchController.scala [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import play.api.mvc.ControllerComponents
+
+@Singleton
+case class LanguageSwitchController @Inject()(configuration: Configuration,
+                                              languageUtils: LanguageUtils,
+                                              cc: ControllerComponents,
+                                              appConfig: AppConfig
+                                             ) extends LanguageController(configuration, languageUtils, cc) {
+
+  override def fallbackURL: String = routes.HelloWorldController.helloWorld.url
```
[the remark this answers is not in the document]
Comment (JoPintoPaul) [reply templates/service/app/controllers/LanguageSwitchController.scala]: Updated with call parathenses.

File: templates/service/project/plugins.sbt [hunk 5]
```diff
@@ -15,3 +14,5 @@ addSbtPlugin("uk.gov.hmrc" % "sbt-artifactory" % "$!sbt_artifactory!$")
 addSbtPlugin("uk.gov.hmrc" % "sbt-distributables" % "$!sbt_distributables!$")
 
 addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.7.5")
+
+addSbtPlugin("org.irundaia.sbt" % "sbt-sassify" % "1.4.11")
```
[the remark this answers is not in the document]
Comment (JoPintoPaul) [reply templates/service/project/plugins.sbt:19]: Wrapped in FRONTEND block.

File: templates/service/build.sbt [hunk 6]
```diff
@@ -12,12 +12,18 @@ lazy val microservice = Project(appName, file("."))
     majorVersion                     := 0,
     scalaVersion                     := "$!SCALA_VERSION!$",
     libraryDependencies              ++= AppDependencies.compile ++ AppDependencies.test,
+    TwirlKeys.templateImports ++= Seq(
```
[the remark this answers is not in the document]
Comment (JoPintoPaul) [reply templates/service/build.sbt:16]: Wrapped in FRONTEND block.

File: templates/service/conf/messages [hunk 7]
```diff
@@ -0,0 +1,11 @@
+service.name = $!APP_NAME!$
```
[the remark this answers is not in the document]
Comment (JoPintoPaul) [reply templates/service/conf/messages:1]: My thinking was that the unit tests show that cookie setting is correctly happening and we can therefore trust that our switching mechanism works. However, I was reluctant to add Welsh messages as I was concerned that Welsh messages could leak into Production without developers realising that they were out of date or no longer reflected the English. 

However, if you think that having a LanguageSwitchController without Welsh messages doesn't make sense, then I can add them.

[the remark this answers is not in the document]
Comment (nigelhp) [reply templates/service/conf/messages:1]: Am happy to leave the decision to PlatUI ;-) - I just wanted to ensure that the issue had been considered.

But does this mean that every service that opts to support Welsh will need to work out what the Welsh equivalents of the standard footer text is?  Presumably we already have these.  Will service teams know where to get them from?

[the remark this answers is not in the document]
Comment (JoPintoPaul) [reply templates/service/conf/messages:1]: Good point re: footer links. I've now added in a messages.cy file with the footer translations as suggested.

[the remark this answers is not in the document]
Comment (nigelhp) [reply templates/service/conf/messages:1]: OK.  I'm afraid you'll also need to add that file to the BACKEND.delete list.

[the remark this answers is not in the document]
Comment (JoPintoPaul) [reply templates/service/conf/messages:1]: Done.

**Pull request merged**